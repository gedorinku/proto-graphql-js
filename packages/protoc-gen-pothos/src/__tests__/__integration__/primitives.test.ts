import { testSchemaGeneration } from "../__helpers__/process.test.helper";

testSchemaGeneration("primitives", "ts-proto", {
  schemaTests: [
    [
      "tests query",
      {
        extraSchema: `
          import { builder } from "./builder";
          import { Message$Ref } from "./testapis/primitives/primitives.pb.pothos";
          import { Message } from "@testapis/ts-proto/lib/testapis/primitives/primitives";

          builder.queryField("test", (t) => t.field({
            type: Message$Ref,
            resolve() {
              return Message.fromPartial({
                requiredPrimitives: {
                  requiredDoubleValue: 2.4,
                  requiredFloatValue: 3.5,
                  requiredInt32Value: 2,
                  requiredInt64Value: "4",
                  requiredUint32Value: 5,
                  requiredUint64Value: "6",
                  requiredSint32Value: 7,
                  requiredSint64Value: "8",
                  requiredFixed32Value: 9,
                  requiredFixed64Value: "10",
                  requiredSfixed32Value: 11,
                  requiredSfixed64Value: "12",
                  requiredBoolValue: true,
                  requiredStringValue: "foobar",
                },
                requiredPrimitivesList: [],
              });
            },
          }));
      `,
        testQuery: `
          query Test {
            test {
              requiredPrimitives {
                ...Primitives
              }
              optionalPrimitives {
                ...Primitives
              }
              requiredPrimitivesList {
                ...Primitives
              }
              optionalPrimitivesList {
                ...Primitives
              }
            }
          }
          fragment Primitives on Primitives {
            requiredDoubleValue
            requiredFloatValue
            requiredInt32Value
            requiredInt64Value
            requiredUint32Value
            requiredUint64Value
            requiredSint32Value
            requiredSint64Value
            requiredFixed32Value
            requiredFixed64Value
            requiredSfixed32Value
            requiredSfixed64Value
            requiredBoolValue
            requiredStringValue
          }
      `,
      },
    ],
  ],
});

testSchemaGeneration("primitives", "ts-proto-with-forcelong-number", {
  genParams: { longNumber: "Int" },
  schemaTests: [
    [
      "tests query",
      {
        extraSchema: `
          import { builder } from "./builder";
          import { Message$Ref } from "./testapis/primitives/primitives.pb.pothos";
          import { Message } from "@testapis/ts-proto-with-forcelong-number/lib/testapis/primitives/primitives";

          builder.queryField("test", (t) => t.field({
            type: Message$Ref,
            resolve() {
              return Message.fromPartial({
                requiredPrimitives: {
                  requiredDoubleValue: 2.4,
                  requiredFloatValue: 3.5,
                  requiredInt32Value: 2,
                  requiredInt64Value: 4,
                  requiredUint32Value: 5,
                  requiredUint64Value: 6,
                  requiredSint32Value: 7,
                  requiredSint64Value: 8,
                  requiredFixed32Value: 9,
                  requiredFixed64Value: 10,
                  requiredSfixed32Value: 11,
                  requiredSfixed64Value: 12,
                  requiredBoolValue: true,
                  requiredStringValue: "foobar",
                },
                requiredPrimitivesList: [],
              });
            },
          }));
      `,
        testQuery: `
          query Test {
            test {
              requiredPrimitives {
                ...Primitives
              }
              optionalPrimitives {
                ...Primitives
              }
              requiredPrimitivesList {
                ...Primitives
              }
              optionalPrimitivesList {
                ...Primitives
              }
            }
          }
          fragment Primitives on Primitives {
            requiredDoubleValue
            requiredFloatValue
            requiredInt32Value
            requiredInt64Value
            requiredUint32Value
            requiredUint64Value
            requiredSint32Value
            requiredSint64Value
            requiredFixed32Value
            requiredFixed64Value
            requiredSfixed32Value
            requiredSfixed64Value
            requiredBoolValue
            requiredStringValue
          }
      `,
      },
    ],
  ],
});
