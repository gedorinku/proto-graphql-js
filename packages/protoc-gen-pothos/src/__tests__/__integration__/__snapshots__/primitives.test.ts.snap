// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema generation with ts-proto tests query 1`] = `
Object {
  "data": Object {
    "test": Object {
      "optionalPrimitives": null,
      "optionalPrimitivesList": Array [],
      "requiredPrimitives": Object {
        "requiredBoolValue": true,
        "requiredDoubleValue": 2.4,
        "requiredFixed32Value": 9,
        "requiredFixed64Value": "10",
        "requiredFloatValue": 3.5,
        "requiredInt32Value": 2,
        "requiredInt64Value": "4",
        "requiredSfixed32Value": 11,
        "requiredSfixed64Value": "12",
        "requiredSint32Value": 7,
        "requiredSint64Value": "8",
        "requiredStringValue": "foobar",
        "requiredUint32Value": 5,
        "requiredUint64Value": "6",
      },
      "requiredPrimitivesList": Array [],
    },
  },
}
`;

exports[`schema generation with ts-proto tests query: schema.graphql 1`] = `
"type Message {
  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitives: Primitives

  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitivesList: [Primitives!]

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitives: Primitives!

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitivesList: [Primitives!]!
}

input MessageInput {
  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitives: PrimitivesInput

  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitivesList: [PrimitivesInput!]

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitives: PrimitivesInput!

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitivesList: [PrimitivesInput!]!
}

type Primitives {
  requiredBoolValue: Boolean!
  requiredBoolValues: [Boolean!]!
  requiredDoubleValue: Float!
  requiredDoubleValues: [Float!]!
  requiredFixed32Value: Int!
  requiredFixed32Values: [Int!]!
  requiredFixed64Value: String!
  requiredFixed64Values: [String!]!
  requiredFloatValue: Float!
  requiredFloatValues: [Float!]!
  requiredInt32Value: Int!
  requiredInt32Values: [Int!]!
  requiredInt64Value: String!
  requiredInt64Values: [String!]!
  requiredSfixed32Value: Int!
  requiredSfixed32Values: [Int!]!
  requiredSfixed64Value: String!
  requiredSfixed64Values: [String!]!
  requiredSint32Value: Int!
  requiredSint32Values: [Int!]!
  requiredSint64Value: String!
  requiredSint64Values: [String!]!
  requiredStringValue: String!
  requiredStringValues: [String!]!
  requiredUint32Value: Int!
  requiredUint32Values: [Int!]!
  requiredUint64Value: String!
  requiredUint64Values: [String!]!
}

input PrimitivesInput {
  requiredBoolValue: Boolean!
  requiredBoolValues: [Boolean!]!
  requiredDoubleValue: Float!
  requiredDoubleValues: [Float!]!
  requiredFixed32Value: Int!
  requiredFixed32Values: [Int!]!
  requiredFixed64Value: String!
  requiredFixed64Values: [String!]!
  requiredFloatValue: Float!
  requiredFloatValues: [Float!]!
  requiredInt32Value: Int!
  requiredInt32Values: [Int!]!
  requiredInt64Value: String!
  requiredInt64Values: [String!]!
  requiredSfixed32Value: Int!
  requiredSfixed32Values: [Int!]!
  requiredSfixed64Value: String!
  requiredSfixed64Values: [String!]!
  requiredSint32Value: Int!
  requiredSint32Values: [Int!]!
  requiredSint64Value: String!
  requiredSint64Values: [String!]!
  requiredStringValue: String!
  requiredStringValues: [String!]!
  requiredUint32Value: Int!
  requiredUint32Values: [Int!]!
  requiredUint64Value: String!
  requiredUint64Values: [String!]!
}

type Query {
  test: Message!
}
"
`;
